src/domain/licence/planstable/PlansTableDialog.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Box, Button, ButtonProps, Dialog, DialogActions, DialogContent, Theme, styled } from '@mui/material';
+import DialogWithGrid from '../../../core/ui/dialog/DialogWithGrid';
+import DialogHeader from '../../../core/ui/dialog/DialogHeader';
+import RouterLink from '../../../core/ui/link/RouterLink';
+import { Caption, CaptionSmall, Text } from '../../../core/ui/typography';
+import GridContainer from '../../../core/ui/grid/GridContainer';
+import GridItem from '../../../core/ui/grid/GridItem';
+import { gutters } from '../../../core/ui/grid/utils';
+import FullWidthButton from '../../../core/ui/button/FullWidthButton';
+import useNavigate from '../../../core/routing/useNavigate';
+import { Plan, getPlanFromId } from './Plan';
+import { usePlanAvailability } from './usePlanAvailability';
+import WrapperMarkdown from '../../../core/ui/markdown/WrapperMarkdown';
+import Gutters from '../../../core/ui/grid/Gutters';
+
+interface PlanTranslation {
+  id: string;
+  name: string;
+  price: string;
+  pricePeriod: string;
+  priceDescription: string;
+  actionButton: string;
+  dialogId?: string;
+  actionUrl?: string;
+  features: string[];
+  disclaimer: string;
+  disabledDisclaimer?: string;
+}
+
+interface DialogTranslation {
+  id: string;
+  title: string;
+  content: string;
+  buttons: {
+    caption: string;
+    variant: string;
+    planId?: string;
+  }[];
+}
+
+const PlanName = styled('h1')(({ theme }) => ({
+  color: theme.palette.primary.main,
+  textAlign: 'center',
+  margin: 0,
+}));
+
+const Price = styled('h1')(({ theme }) => ({
+  color: theme.palette.primary.main,
+}));
+
+const lines = (theme: Theme) => `1px solid ${theme.palette.divider}`;
+
+const buttonVariant = (variant: string): ButtonProps['variant'] => {
+  switch (variant) {
+    case 'text':
+      return 'text';
+    case 'contained':
+      return 'contained';
+    case 'outlined':
+      return 'outlined';
+    default:
+      return undefined;
+  }
+};
+
+interface PlansTableDialogProps {
+  open: boolean;
+  onClose: () => void;
+  onSelectPlan: (plan: Plan) => void;
+}
+
+const PlansTableDialog = ({ open, onClose, onSelectPlan }: PlansTableDialogProps) => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+
+  const { isPlanAvailable } = usePlanAvailability();
+  const [subDialogOpen, setSubDialogOpen] = useState<string>();
```

Comment:
We can think of a better name for the subDialog, something like 'confrimPlanDialog'.